### ISSUE TITLE
INC-546 - Metrics on messages ingested vs processed

### ISSUE THREAD

**Issue by wedamija** [getsentry/sentry#59290]
Action item from https://www.notion.so/sentry/INC-546-Profiling-issue-occurrences-were-no-longer-being-ingested-by-issue-platform-bcb31d75a05c4f6a903c3d6a8224eca5

We want to keep track of the total number of messages ingested, and how many we actually processed. We can then alert on having a high number of unprocessed messages.

**Comment by vartec** [comment]
I think we already have these, `sentry.occurrence_ingest.messages` (counter) and `sentry.occurrence_consumer.process_message` (timer, so `.count` for counter)

https://app.datadoghq.com/notebook/6861386/messages-ingested-vs-processed

**Comment by isabellaenriquez** [comment]
good catch! @wedamija was there something else you had in mind or can we close this?

**Comment by vartec** [comment]
Here's what it looks like during the incident (diff between the two metrics with auto-smoothing):
![Screenshot 2023-11-02 at 4 10 32 PM](https://github.com/getsentry/sentry/assets/2284726/13991bee-a2e4-43b0-9627-7cf55cc2ed0a)

@isabellaenriquez @wedamija I guess it still needs to be added to [our dashboard](https://app.datadoghq.com/dashboard/fax-p33-gf5/issue-platform), smoothing tweaked as necessary and alerting set up.

**Comment by wedamija** [comment]
> entry.occurrence_consumer.process_message

Process message doesn't catch the problem from our incident, but actually we can use `dropped_invalid_payload_type` for that, so yeah other than adding something to a dashboard that'd catch the incident I think we're good!

**Comment by hubertsentry** [comment]
> Process message doesn't catch the problem from our incident,

@wedamija can you elaborate on this? 

I feel special case for `dropped_invalid_payload_type` is not enough for future incidents. 

For this incident, `dropped_invalid_payload_type` might be able to help you capture this issue, but we might still get into different issues messages don't get processed.  I

MHO, having a "capture all" ingested v.s processed monitoring per occurrence type (we don't have per "occurrence type" for `sentry.occurrence_consumer.process_message`) might work better and could be more future proof.
